Add support for pyflakes 3.1 and pycodestyle 2.11

yan12125 opened the pull request:
`ContinueInFinally` is dropped in pyflakes 3.1 [1].

flake8 is bumped to a version that supports pyflakes 3.1 [2].

Closes https://github.com/python-lsp/python-lsp-server/issues/412

[1] https://github.com/PyCQA/pyflakes/pull/752
[2] https://flake8.pycqa.org/en/latest/release-notes/6.1.0.html

yan12125 committed:
Add support for pyflakes 3.1

`ContinueInFinally` is dropped in pyflakes 3.1 [1].

flake8 is bumped to a version that supports pyflakes 3.1 [2].

pyflakes is also bumped to have the same version range as flake8 [2]
(thanks ccordoba12 for the notice).

Closes https://github.com/python-lsp/python-lsp-server/issues/412

[1] https://github.com/PyCQA/pyflakes/pull/752
[2] https://flake8.pycqa.org/en/latest/release-notes/6.1.0.html

Co-authored-by: Carlos Cordoba <ccordoba12@gmail.com>

yan12125 commented:
Forgot to mention - the lower bound for pyflakes is bumped to 3.1.0 as flake8 6.1.0 already requires that version, as described in flake8 release notes.

ccordoba12 commented on pyproject.toml:
```diff
@@ -28,22 +28,22 @@ Homepage = "https://github.com/python-lsp/python-lsp-server"
-    "flake8>=5.0.0,<7",
+    "flake8>=6.1.0,<7",
     "mccabe>=0.7.0,<0.8.0",
     "pycodestyle>=2.9.0,<2.11.0",
```
```suggestion
    "pycodestyle>=2.11.0,<2.12.0",
```

ccordoba12 commented on pyproject.toml:
```diff
@@ -28,22 +28,22 @@ Homepage = "https://github.com/python-lsp/python-lsp-server"
-flake8 = ["flake8>=5.0.0,<7"]
+flake8 = ["flake8>=6.1.0,<7"]
 mccabe = ["mccabe>=0.7.0,<0.8.0"]
 pycodestyle = ["pycodestyle>=2.9.0,<2.11.0"]
```
```suggestion
pycodestyle = ["pycodestyle>=2.11.0,<2.12.0"]
```

ccordoba12 reviewed (commented):
Thanks @yan12125 for your help with this! We also need to bump our supported `pycodestyle` version, according to:

https://flake8.pycqa.org/en/latest/release-notes/6.1.0.html

yan12125 commented:
Good point, thanks!

---

> ERROR: Could not find a version that satisfies the requirement flake8<7,>=6.1.0; extra == "all" (from python-lsp-server[all,test]) (from versions: 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0, 1.1, 1.2, 1.3, 1.3.1, 1.4, 1.5, 1.6, 1.6.1, 1.6.2, 1.7.0, 2.0, 2.1.0, 2.2.0, 2.2.1, 2.2.2, 2.2.3, 2.2.4, 2.2.5, 2.3.0, 2.4.0, 2.4.1, 2.5.0, 2.5.1, 2.5.2, 2.5.3, 2.5.4, 2.5.5, 2.6.0, 2.6.1, 2.6.2, 3.0.0b1, 3.0.0b2, 3.0.0, 3.0.1, 3.0.2.dev0, 3.0.2.dev1, 3.0.2, 3.0.3, 3.0.4, 3.1.0, 3.1.1, 3.2.0, 3.2.1, 3.3.0.dev0, 3.3.0, 3.4.0, 3.4.1, 3.5.0, 3.6.0, 3.7.0, 3.7.1, 3.7.2, 3.7.3, 3.7.4, 3.7.5, 3.7.6, 3.7.7, 3.7.8, 3.7.9, 3.8.0a1, 3.8.0a2, 3.8.0, 3.8.1, 3.8.2, 3.8.3, 3.8.4, 3.9.0, 3.9.1, 3.9.2, 4.0.0, 4.0.1, 5.0.0, 5.0.1, 5.0.2, 5.0.3, 5.0.4)

(https://github.com/python-lsp/python-lsp-server/actions/runs/5839448515/job/15843654307)

New flake8 requires Python 3.8 [1]. Is it better to drop Python 3.7 from python-lsp-server or install older dependencies for older Python?

[1] https://flake8.pycqa.org/en/latest/release-notes/6.0.0.html

ccordoba12 commented:
> New flake8 requires Python 3.8 [1]. Is it better to drop Python 3.7 from python-lsp-server

Right. I already did that in pull request #417. So, please rebase your pull request or merge with the latest master to get the changes. Then, your pull request should pass.

yan12125 commented:
> Right. I already did that in pull request https://github.com/python-lsp/python-lsp-server/pull/417. So, please rebase your pull request or merge with the latest master to get the changes. Then, your pull request should pass.

Thank you! Rebased and squashed.

ccordoba12 reviewed (approved):
Thanks @yan12125 for your help with this!
